### DISCUSSION TITLE
React Desgin Patterns

### DISCUSSION THREAD

**Discussion by vczb** [frontendbr/forum#2125] [Discussão]
Olá, estou fazendo um levantamento sobre Design Patterns para React, no momento consegui listar apenas dois, vocês conhecem outros ?

- Flux
- BLoC

**Comment by wilsonneto-swe** [comment]
Todos os design patterns raíz do GoF, e todos os novos, na verdade todos os design patterns são agnóticos a linguagem ou framework senão não seriam design patterns, então, todos os design patterns podem entrar na lista... Adapter, Observer, Abstract Facory, Factory Method, Facade, Strategy, Commands... Inclusive todos os padrões arquiteturais macro e micro também podem ser usados, repositories, useCases, CQS, arquitetura hexagonal, arquitetura limpa, em camadas e assim por diante. Podendo ir para linhas de pensamentos e conceitos com SOLID, DDD, DRY e etc... 
Flux e BLoC podem ser aplicados no backend, flutter, java e etc, inclusive Flux e SAGASé bastante utilizado em microserviços.

Apenas meus 25 cents, abraços!

**Comment by ninetails** [comment]
Em código, tem uma listinha aqui já:

> https://reactpatterns.com/

Agora, tem padrões mais amplos, menos específicos como:
- para estilização:
  - CSS in JS
  - CSS modules
- para estado/[State Pattern](https://refactoring.guru/design-patterns/state): (uma infinidade e inclusive combinações, mas comum em libs de componentização)
  - Redux
  - Observables
  - Sagas
  - outros reativos
- da própria lib:
  - Context API
  - Hooks & Custom Hooks
  - Suspense (Render as you Fetch)
- da comunidade:
  - [Container & Presentational Components](https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0)

Padrões de projeto (design patterns) tanto de Orientação a Objeto (Gang of Four) quanto de [Programação Funcional](https://fsharpforfunandprofit.com/fppatterns/)...

E tem coisa que vai depender do ecossistema que será a base. O que vc vai plugar pro React.... etc

**Comment by vczb** [comment]
@ninetails @wilsonneto-dev Valeu pessoal :)

**Comment by nazarepiedady** [comment]
É possível que ajude também: https://alexkondov.com/tao-of-react/
